app/services/foreman_ansible/variables_importer.rb

```diff
@@ -37,6 +37,34 @@ def import_variables(role_variables, new_roles)
       )
     end
 
+    def import_variables_roles(roles)
+      if roles['new']
+        imported_roles = roles['new'].keys
+        variables_to_import = {}
+        remote_variables.each do |role, variables|
+          next unless imported_roles.include? role
+          role_obj = AnsibleRole.find_by(:name => role)
+          variables_to_import[role] = {}
+          values = initialize_variables(variables, role_obj)
+          values.each do |value|
+            variables_to_import[role][value.key] = value.attributes.to_json
+          end
+        end
+        create_new_variables(variables_to_import)
+      end
+      return if roles['old'].empty?
+      variables_to_update = { 'new' => {}, 'obsolete' => {}, 'update' => {} }
+      import_variable_names([]).each do |kind, variables|
+        variables.each do |variable|
+          if roles['old'].values.map { |role| role['id'] }.include?(variable.ansible_role_id)
+            role = AnsibleRole.where(:id => variable.ansible_role_id)[0]
+            variables_to_update[kind][role.name] = { variable.key => variable.attributes.to_json }
```

Comment:
This assigns variable into hash under `role.name`, but what if there are multiple variables that pass the condition above?